### ISSUE TITLE
Unable to rejoin a group when consumer poll timeout in `confluent-kafka-go v2.1.0`

### ISSUE THREAD

**Issue by ljy201758505108** [confluentinc/confluent-kafka-go#980]
Description
===========
When I upgraded `confluent-kafka-go` from `v2.0.2` to `v2.1.0`, it was not possible to rejoin consumers back to the consumer group in `2.1.0` when the time interval between two consumer `poll` operations exceeded `max.poll.interval.ms`, but it was feasible in `v2.0.2`.

When Group myTopicName `join state` changed `wait-unassign-to-complete -> init`:

`v2.0.2`: Group myTopicName join with 1 subscribed topic(s)...

`v2.1.0`: Requseting metadata for 1/1 topics: periodic topic and broker list refresh...

client config:

```
request.timeout.ms: 60000
retries: 5
socket.timeout.ms: 60000
```

consumer config:

```
session.timeout.ms: 45000
max.poll.interval.ms: 60000
heartbeat.interval.ms: 10000
socket.timeout.ms: 60000
enable.auto.commit: false
enable.auto.offset.store: false
group.id: myTopicName
```

How to reproduce
================

pseudo-code：

```
c,err := NewConsumer(..)
c.subscribe(topic,nil)
for {
	ev := c.poll(100)
	time.sleep(65000) // set `sleep` 65s to make `c.poll` timeout
}
```

Checklist
=========
Please provide the following information:

 - [√] confluent-kafka-go and librdkafka version (`v2.0.2` to `v2.1.0`):
 - [√] Apache Kafka broker version: 2.3.0
 - [√] Client configuration: `ConfigMap{...}`
 - [√] Operating system:linux x86_64
 - [√] Provide client logs (with `"debug": ".."` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by ljy201758505108** [comment]
> By any chance, are you using "go.logs.channel.enable"? There was a recent bug fixed around this, earlier max.poll.interval.ms was not honored while using this property.
> 
> I can't reproduce it locally given the pseudocode, it seems to work fine for me (I can rejoin the group).
Thank for your reply. I'm not using “go.logs.channel.enable”, I tried using this parameter in my test code, but the results were the same as before. None of them rejoined the group after the error, but the consumer is still polling empty messages. When I looked at the group, there were no active consumers.

**Comment by ljy201758505108** [comment]
Here are my test code ：
`package main

import (
	"fmt"
	"os"
	"time"

	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
)

func main() {
	bootstrapServers := "xxxx:xx"
	group := "dGVzdDAz"
	c, err := kafka.NewConsumer(&kafka.ConfigMap{
		"bootstrap.servers":        bootstrapServers,
		"group.id":                 group,
		"heartbeat.interval.ms":    100,
		"session.timeout.ms":       45000,
		"max.poll.interval.ms":     60000,
		"enable.auto.commit":       false,
		"auto.offset.reset":        "earliest",
		"enable.auto.offset.store": false,

		"security.protocol":                   "sasl_ssl",
		"sasl.mechanism":                      "PLAIN",
		"sasl.username":                       "xxxx",
		"sasl.password":                       "xxxx",
		"debug":                               "all",
	})
	if err != nil {
		fmt.Printf("err: %v\n", err)
		os.Exit(1)
	}
	c.Subscribe("dGVzdDAz", nil)

	run := true

	for run {
		ev := c.Poll(100)
		fmt.Printf("Message Received:%v\n", ev)
		time.Sleep(65 * time.Second)
	}

	fmt.Printf("Closing consumer\n")
	c.Close()
}`

**Comment by haoxins** [comment]
I'm facing the same issue when I upgraded from 2.0.2 to 2.1.0.

**Comment by milindl** [comment]
Thanks for this report. I was able to reproduce it locally as well, I'll look into it further.

**Comment by marcosbmf** [comment]
+1 having the same issue when upgraded from 2.0.2 to 2.1.0.
On the 2.0.2 my workers would run fine, now they exit the group and don't rejoin with the following error:

```
%4|1681402127.003|MAXPOLL|rdkafka#consumer-1| [thrd:main]: Application maximum poll interval (20000ms) exceeded by 264ms (adjust max.poll.interval.ms for long-running message processing): leaving group
```

**Comment by jadamscn** [comment]
I am also getting this, despite poll() definitely being called. 

Steps to reproduce:
1. Create a new empty topic
2. Have the consumer join a group and subscribe to that topic
3. Run poll in a for loop - e.g. for { c.Poll(100) }
4. Do not send any messages to the topic for the duration of max.poll.interval.ms

Despite poll being called (and logs showing the fetch), the consumer is still kicked out of the group after max.poll.interval.ms.

**Comment by fkarakas** [comment]
+1 same issue. Where is @edenhill ?

**Comment by milindl** [comment]
This is a librdkafka issue, we're fixing it here: https://github.com/confluentinc/librdkafka/pull/4256

**Comment by fkarakas** [comment]
ok thanks @milindl but what I don't understand is that we should have detected this kind of problem easily with a basic test in the confluent go lib, no ?

**Comment by emasab** [comment]
@fkarakas about Magnus, check [this](https://github.com/confluentinc/librdkafka/issues/4212#issuecomment-1463704171) comment.

**Comment by milindl** [comment]
And for the tests, yes, we should have ideally detected it.
We'll add a test once we fix the issue, since the current suite (and the much larger librdkafka suite) passed despite this issue.

**Comment by fkarakas** [comment]
OK.
But it is always preferable to have a falling test before the fix, so when the fix is done and the test passes you are sure that your test and the fix are correct :)

**Comment by milindl** [comment]
@fkarakas , yep, that's what we did, we added a failing test to librdkafka, where the actual issue exists. I was talking about extending the go test suite to include more tests, too.

Fix is available in the v2.1.1-RC1 and we expect to have it in v2.1.1 when it is released after some soak testing.
